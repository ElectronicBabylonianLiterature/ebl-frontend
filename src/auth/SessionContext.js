import React from 'react'
import Session from './Session'

const SessionContext = React.createContext(new Session('', '', 0, []))

export default SessionContext
