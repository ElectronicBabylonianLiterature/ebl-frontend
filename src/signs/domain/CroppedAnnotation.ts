import { MesopotamianDate } from 'chronology/domain/Date'

type base64String = string

export interface CroppedAnnotation {
  image: base64String
  fragmentNumber: string
  provenance?: string
  script: string
  label: string
  date?: MesopotamianDate
}
