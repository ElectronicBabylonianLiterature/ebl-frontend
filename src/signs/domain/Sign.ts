export default interface Sign {
  name: string
  lists: readonly SignListRecord[]
  values: readonly Value[] | null
  logograms: readonly Logogram[]
  mesZl: string | null
}
interface Logogram {
  logogram: string
  atf: string
  wordId: readonly string[]
  schrammLogogramme: string
}
interface Value {
  value: string
  subIndex: number | null
}

interface SignListRecord {
  name: string
  number: string
}

export interface SignQuery {
  value: string
  subIndex: string
  isIncludeHomophones: boolean
  isComposite: boolean
  signList: string
}
