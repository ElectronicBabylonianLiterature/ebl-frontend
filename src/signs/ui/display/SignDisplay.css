.signDisplay__LogogramInfo {
  min-width: 25%;
}

.signDisplay__FontInfo {
  font-size: 0.8em;
  color: grey;
}

.CuneiformFonts__heading-old-babylonian-monumental {
  font-family: 'Old-Babylonian (Monumental)', 'Adobe Blank', Junicode, serif;
}

.CuneiformFonts__heading-old-babylonian-literature {
  font-family: 'Old-Babylonian (Literature)', 'Adobe Blank', Junicode, serif;
}

.CuneiformFonts__heading-old-babylonian-cursive {
  font-family: 'Old-Babylonian Cursive', 'Adobe Blank', Junicode, serif;
}

.CuneiformFonts__heading-hittite {
  font-family: Hittite, 'Adobe Blank', Junicode, serif;
}

.CuneiformFonts__heading-neo-babylonian {
  font-family: Neo-Babylonian, 'Adobe Blank', Junicode, serif;
}

@font-face {
  font-display: swap;
  font-family: 'Adobe Blank';
  src: url('fonts/AdobeBlank.otf.woff') format('woff');
}

@font-face {
  font-display: swap;
  font-family: 'Old-Babylonian (Monumental)';
  src: url('fonts/SantakkuM.woff') format('woff');
}

@font-face {
  font-display: swap;
  font-family: 'Old-Babylonian (Literature)';
  src: url('fonts/OBFreie.woff') format('woff');
}

@font-face {
  font-display: swap;
  font-family: 'Old-Babylonian Cursive';
  src: url('fonts/Santakku.woff') format('woff');
}

@font-face {
  font-display: swap;
  font-family: Hittite;
  src: url('fonts/UllikummiA.woff') format('woff');
}

@font-face {
  font-display: swap;
  font-family: Neo-Babylonian;
  src: url('fonts/Esagil.woff') format('woff');
}
