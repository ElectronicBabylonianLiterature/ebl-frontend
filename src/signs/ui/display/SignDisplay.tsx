import React from 'react'
import { Col, Row } from 'react-bootstrap'

export default function SignDisplay(): JSX.Element {
  return (
    <Row className="mt-5">
      <Col className="text-center mt-5">Under Construction</Col>
    </Row>
  )
}
