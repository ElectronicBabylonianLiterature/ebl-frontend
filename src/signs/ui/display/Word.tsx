import withData, { WithoutData } from 'http/withData'
import Word from 'dictionary/domain/Word'
import React from 'react'
import WordService from 'dictionary/application/WordService'
import { Lemma } from 'dictionary/ui/search/Word'

type Props = {
  data: Word
  wordService: WordService
}

export default withData<WithoutData<Props>, { wordId; wordService }, Word>(
  ({ data }) => <Lemma word={data} />,
  (props) => props.wordService.find(props.wordId)
)
