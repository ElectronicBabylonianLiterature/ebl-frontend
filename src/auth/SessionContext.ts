import React, { Context } from 'react'
import { guestSession, Session } from 'auth/Session'

const SessionContext: Context<Session> = React.createContext<Session>(
  guestSession
)

export default SessionContext
