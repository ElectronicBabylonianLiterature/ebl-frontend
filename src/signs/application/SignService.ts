import Bluebird from 'bluebird'
import Sign, { SignQuery } from 'signs/domain/Sign'
import SignRepository from 'signs/infrastructure/SignRepository'
import { AnnotationToken } from 'fragmentarium/domain/annotation-token'

import { CroppedAnnotation } from 'signs/domain/CroppedAnnotation'

export default class SignService {
  private readonly signsRepository: SignRepository

  constructor(signsRepository: SignRepository) {
    this.signsRepository = signsRepository
  }

  getImages(signName: string): Bluebird<CroppedAnnotation[]> {
    return this.signsRepository.getImages(signName)
  }

  associateSigns(
    tokens: ReadonlyArray<ReadonlyArray<AnnotationToken>>
  ): Bluebird<ReadonlyArray<ReadonlyArray<AnnotationToken>>> {
    return this.signsRepository.associateSigns(tokens)
  }

  search(signQuery: SignQuery): Bluebird<Sign[]> {
    return this.signsRepository.search(signQuery)
  }

  find(signName: string): Bluebird<Sign> {
    return this.signsRepository.find(signName)
  }
}
