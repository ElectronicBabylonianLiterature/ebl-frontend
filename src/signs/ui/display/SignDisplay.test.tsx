import React, { ReactNode } from 'react'
import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route, RouteComponentProps } from 'react-router-dom'
import SessionContext from 'auth/SessionContext'
import SignDisplay from 'signs/ui/display/SignDisplay'
import MemorySession from 'auth/Session'
import SignService from 'signs/application/SignService'
import Bluebird from 'bluebird'
import Sign, { Value } from 'signs/domain/Sign'
import WordService from 'dictionary/application/WordService'
import Word from 'dictionary/domain/Word'
import { CroppedAnnotation } from 'signs/domain/CroppedAnnotation'

jest.mock('signs/application/SignService')
jest.mock('dictionary/application/WordService')
const signService = new (SignService as jest.Mock<jest.Mocked<SignService>>)()
const wordService = new (WordService as jest.Mock<jest.Mocked<WordService>>)()
const session = new MemorySession(['read:words'])
const sign = new Sign({
  lists: [],
  logograms: [],
  name: 'BU',
  unicode: [74127],
  values: [new Value('gabu'), new Value('dul', 10), new Value('du', 1)],
  mesZl: `123	**ALSK13**	𒁇𒍴
Lorem ipsum dolor *sit* amet, consetetur <span style="color: #00610F;">*sadipscing*(*l*)*ubasd*</span>sadipscing elitr, sed diam *nonumy*
eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet c
lita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
^aba^𒁇𒊩 = BAR-MUNUS = *parratu*, weibliches Lamm. CAD <span style="color: #00610F;">P</span> 192b liest *parsallu*.
<span style="color: #00610F;">P</span>
^aba^𒁇𒋝
^asdg^𒋝
^ghas^𒁈
𒁖^asd^
𒍴^q12asd^
`,
  LaBaSi: '123',
})

const word: Word = {
  guideWord: '',
  arabicGuideWord: '',
  cdaAddenda: '',
  homonym: '',
  lemma: [],
  oraccWords: [],
  akkadischeGlossareUndIndices: [],
  pos: [],
  _id: '',
}

const croppedAnnotation: CroppedAnnotation = {
  image: 'test-base64-string',
  fragmentNumber: '',
  script: 'NA',
  label: "i stone wig 1'",
}

let container: HTMLElement

function renderSignDisplay(signName: string) {
  container = render(
    <MemoryRouter initialEntries={[`/signs/${signName}`]}>
      <SessionContext.Provider value={session}>
        <Route
          path="/signs/:id"
          render={(props: RouteComponentProps<{ id: string }>): ReactNode => (
            <SignDisplay
              wordService={wordService}
              signService={signService}
              {...props}
            />
          )}
        />
      </SessionContext.Provider>
    </MemoryRouter>
  ).container
}

describe('Sign Display', () => {
  beforeEach(async () => {
    signService.search.mockReturnValue(Bluebird.resolve([]))
    signService.getImages.mockReturnValue(Bluebird.resolve([croppedAnnotation]))
    signService.find.mockReturnValue(Bluebird.resolve(sign))
    wordService.find.mockReturnValue(Bluebird.resolve(word))
    renderSignDisplay(sign.name)
    await screen.findAllByText(sign.name)
    expect(signService.find).toBeCalledWith(sign.name)
  })
  it('Sign Display Snapshot', async () => {
    expect(container).toMatchSnapshot()
  })
})
