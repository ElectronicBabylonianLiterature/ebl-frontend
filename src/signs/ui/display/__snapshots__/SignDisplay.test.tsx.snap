// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Display Sign Display Snapshot 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="main__content"
    >
      <header
        class="main__header"
      >
        <nav
          aria-label="breadcrumb"
          class="main__breadcrumbs"
        >
          <ol
            class="breadcrumb"
          >
            <li
              class="breadcrumb-item active"
            >
              <a
                href="/"
              >
                eBL
              </a>
            </li>
            <li
              class="breadcrumb-item active"
            >
              <a
                href="/signs"
              >
                Signs
              </a>
            </li>
            <li
              aria-current="page"
              class="breadcrumb-item active"
            >
              BU
            </li>
          </ol>
        </nav>
        <h2
          class="main__heading"
        >
           
        </h2>
        <div
          class="main__toolbar btn-toolbar"
          role="toolbar"
        />
      </header>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-3"
          >
            <h2>
              BU
            </h2>
          </div>
          <div
            class="col"
          >
            <div
              class="row"
            >
              <div
                class="col-4"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto"
                  >
                    <h3
                      class="CuneiformFonts__heading-old-babylonian-monumental"
                    >
                      íÜè
                    </h3>
                  </div>
                  <div
                    class="col"
                  >
                    <span
                      class="text-secondary signHeading__cuneiformSign__description"
                    >
                      Old Babylonian (Monumental)
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="col-4"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto"
                  >
                    <h3
                      class="CuneiformFonts__heading-old-babylonian-cursive"
                    >
                      íÜè
                    </h3>
                  </div>
                  <div
                    class="col"
                  >
                    <span
                      class="text-secondary signHeading__cuneiformSign__description"
                    >
                      Old Babylonian Cursive
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="col-4"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto"
                  >
                    <h3
                      class="CuneiformFonts__heading-hittite"
                    >
                      íÜè
                    </h3>
                  </div>
                  <div
                    class="col"
                  >
                    <span
                      class="text-secondary signHeading__cuneiformSign__description"
                    >
                      Hittite
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="col-4"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto"
                  >
                    <h3
                      class=""
                    >
                      íÜè
                    </h3>
                  </div>
                  <div
                    class="col"
                  >
                    <span
                      class="text-secondary signHeading__cuneiformSign__description"
                    >
                      Neo-Assyrian
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="col-4"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-auto"
                  >
                    <h3
                      class="CuneiformFonts__heading-neo-babylonian"
                    >
                      íÜè
                    </h3>
                  </div>
                  <div
                    class="col"
                  >
                    <span
                      class="text-secondary signHeading__cuneiformSign__description"
                    >
                      Neo-Babylonian
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="col"
              >
                <i
                  class="fas fa-info-circle"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h3>
                  ‚Ö†. Sign Information
                </h3>
              </div>
            </div>
            <div
              class="ml-5"
            >
              <div
                class="row"
              >
                <div
                  class="col-auto"
                >
                  Sign Lists:
                   
                </div>
                <i
                  class="fas fa-info-circle"
                />
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <span
                    class="mr-3"
                  >
                    <strong>
                      La
                    </strong>
                    te
                    <strong>
                       Ba
                    </strong>
                    bylonian
                    <strong>
                       Si
                    </strong>
                    gns: 
                    123
                  </span>
                  <a
                    class="text-dark "
                    href="https://labasi.acdh.oeaw.ac.at/tablets/sign/detail/123"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      class="fas fa-external-link-square-alt fa-1x"
                    />
                  </a>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Readings: 
                  <em>
                    du
                  </em>
                  , 
                  <em>
                    dul
                  </em>
                  ‚ÇÅ‚ÇÄ, 
                  <em>
                    gabu
                  </em>
                  ~x~
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mt-5 row"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h3>
                ‚Ö°. Mesopotamisches Zeichenlexikon
              </h3>
            </div>
          </div>
          <div
            class="ml-5"
          >
            <div
              class="row"
            >
              <div
                class="ml-4 mt-3 col"
              >
                <div
                  class="mesZL__popover"
                >
                  <pre>
                    <div
                      class="text-center"
                    >
                      123	
                      <strong>
                        ALSK13
                      </strong>
                      	íÅáíç¥
                    </div>
                  </pre>
                </div>
                <div
                  class="text-align-justify"
                >
                  Lorem ipsum dolor 
                  <em>
                    sit
                  </em>
                   amet, consetetur 
                  <span
                    style="color: #00610F;"
                  >
                    <em>
                      sadipscing
                    </em>
                    (
                    <em>
                      l
                    </em>
                    )
                    <em>
                      ubasd
                    </em>
                  </span>
                  sadipscing elitr, sed diam 
                  <em>
                    nonumy
                  </em>
                  <p />
                  

                  <p>
                    eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet c
                  </p>
                  

                  <p>
                    lita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
                  </p>
                  

                  <p>
                    <sup>
                      aba
                    </sup>
                    íÅáíä© = BAR-MUNUS = 
                    <em>
                      parratu
                    </em>
                    , weibliches Lamm. CAD 
                    <span
                      style="color: #00610F;"
                    >
                      P
                    </span>
                     192b liest 
                    <em>
                      parsallu
                    </em>
                    .
                  </p>
                  

                  <p>
                    <span
                      style="color: #00610F;"
                    >
                      P
                    </span>
                  </p>
                  

                  <p>
                    <sup>
                      aba
                    </sup>
                    íÅáíãù
                  </p>
                  

                  <p>
                    <sup>
                      asdg
                    </sup>
                    íãù
                  </p>
                  

                  <p>
                    <sup>
                      ghas
                    </sup>
                    íÅà
                  </p>
                  

                  <p>
                    íÅñ
                    <sup>
                      asd
                    </sup>
                  </p>
                  

                  <p>
                    íç¥
                    <sup>
                      q12asd
                    </sup>
                  </p>
                </div>
                <div
                  class="text-center border border-dark mx-5 m-3 p-2"
                >
                  <strong>
                    From
                  </strong>
                  <br />
                  Borger, R.
                  , 
                  <em>
                    Mesopotamisches Zeichenlexikon. Zweite, revidierte und aktualisierte Auflage
                  </em>
                  . 
                  Alter Orient und Altes Testament 305. M√ºnster: Ugarit-Verlag, ¬≤2010; Kapitel ‚Ö¢
                  .
                  <br />
                  <strong>
                    By permission from Ugarit-Verlag
                  </strong>
                  <br />
                  <a
                    class="text-dark "
                    href="https://ugarit-verlag.com/en/products/0e8e7ca5d1f5493aa351e3ebc42fb514"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      class="pointer__hover my-2 fas fa-shopping-cart fa-2x"
                    />
                    \`
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <figure
                class="figure"
              >
                <img
                  class="sign-images__sign-image figure-img img-fluid"
                  src="data:image/png;base64, test-base64-string"
                />
                <figcaption
                  class="figure-caption"
                >
                  <a
                    href="/fragmentarium/"
                  />
                  i stone wig 1' 
                  ¬†(
                  NA
                  )
                </figcaption>
              </figure>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col offset-5"
            >
              <ul
                class="pagination"
              >
                <li
                  class="page-item active"
                >
                  <span
                    class="page-link"
                  >
                    1
                    <span
                      class="sr-only"
                    >
                      (current)
                    </span>
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
